modules/apps/roles/roles-admin-test/src/testIntegration/java/com/liferay/roles/admin/internal/exportimport/data/handler/test/RolesAdminPortletDataHandlerTest.java

```diff
@@ -97,10 +96,13 @@ protected void checkManifestSummaryReferrerClassNames(
 		for (String manifestSummaryKey :
 				manifestSummary.getManifestSummaryKeys()) {
 
-			Stream<String> classNamesStream = classNames.stream();
+			for (String className : classNames) {
+				if (manifestSummaryKey.endsWith(className)) {
+					Assert.assertTrue(manifestSummaryKey.endsWith(className));
```

Comment:
@jeffwu0724 You can do this: directly return in the `if` body; `Assert.fail()` if the loop finishes.

Also, the outer loop doesn't make sense either... it loops through manifest summary keys, but if the first key doesn't "anyMatch", the outer loop will exit... Please double check its logic. Maybe the `Assert.fail()` should be moved out of the outer loop?

Reply:
@dantewang 
Since in the original logic, it will pass this test only if every manifestSummaryKey has one or more that satisfied manifestSummaryKeyendsWith(classNamesStream)
If we use return in this case, after the first one that match, we will return, which did not cover every manifestSummaryKey, only the first satisfied manifestSummaryKey.
If I tried to break in the inner loop when I found something match, it makes no sense because at the end it will break if I do not find anything match.
If I put it outside the outer loop,  then we still need to return inside to make it works, and we need to make sure all the manifestSummaryKey get covered too.
Therefore, after discussed with Tina, we think using boolean flag may be a better option